File: fix.py
import re

with open('data.js', 'r', encoding='utf-8') as f:
    data = f.read()

# Update DEFAULT_DEPARTMENTS array
data = re.sub(
    r'const DEFAULT_DEPARTMENTS = \[.*\];',
    'const DEFAULT_DEPARTMENTS = ["CSE", "IT", "AIDS", "AIML", "Cyber Security", "IoT", "ECE", "EEE", "Mechanical", "Civil"];',
    data
)

# Replace dept values
data = data.replace('dept: "CYS"', 'dept: "Cyber Security"')
data = data.replace('dept: "IOT"', 'dept: "IoT"')
data = data.replace('dept: "ME"', 'dept: "Mechanical"')
data = data.replace('dept: "CE"', 'dept: "Civil"')

# Replace class section names
data = re.sub(r'"CYS (\d[a-z]{2} Year)"', r'"Cyber Security \1"', data)
data = re.sub(r'"IOT (\d[a-z]{2} Year)"', r'"IoT \1"', data)
data = re.sub(r'"ME (\d[a-z]{2} Year)"', r'"Mechanical \1"', data)
data = re.sub(r'"CE (\d[a-z]{2} Year)"', r'"Civil \1"', data)

with open('data.js', 'w', encoding='utf-8') as f:
    f.write(data)

print("data.js updated")

File: fix.js
const fs = require('fs');
let data = fs.readFileSync('data.js', 'utf8');

// Update DEFAULT_DEPARTMENTS array
data = data.replace(
  /const DEFAULT_DEPARTMENTS = \[.*\];/,
  'const DEFAULT_DEPARTMENTS = ["CSE", "IT", "AIDS", "AIML", "Cyber Security", "IoT", "ECE", "EEE", "Mechanical", "Civil"];'
);

// Replace dept values
data = data.replace(/dept: "CYS"/g, 'dept: "Cyber Security"');
data = data.replace(/dept: "IOT"/g, 'dept: "IoT"');
data = data.replace(/dept: "ME"/g, 'dept: "Mechanical"');
data = data.replace(/dept: "CE"/g, 'dept: "Civil"');

// Replace class section names
data = data.replace(/"CYS (\d[a-z]{2} Year)"/g, '"Cyber Security $1"');
data = data.replace(/"IoT (\d[a-z]{2} Year)"/g, '"IoT $1"');
data = data.replace(/"ME (\d[a-z]{2} Year)"/g, '"Mechanical $1"');
data = data.replace(/"CE (\d[a-z]{2} Year)"/g, '"Civil $1"');

fs.writeFileSync('data.js', data);
console.log("data.js updated");
